### PR TITLE
openssh: Fix pre-authentication remote code execution in sshd.

### PR DESCRIPTION
**Pull request by brooksdavis**
Reported by:	Qualys Threat Research Unit (TRU)
Approved by:	so
Security:	FreeBSD-SA-24:04.openssh
Security:	CVE-2024-6387

Change sshd version string from FreeBSD to CheriBSD as what we're publishing doesn't match FreeBSD-20240701.

(cherry picked from commit 2abea9df01655633aabbb9bf3204c90722001202)

### CODE REVIEW

File: crypto/openssh/version.h [hunk 1]
```diff
@@ -5,4 +5,4 @@
 #define SSH_PORTABLE	"p1"
 #define SSH_RELEASE	SSH_VERSION SSH_PORTABLE
 
-#define SSH_VERSION_FREEBSD	"FreeBSD-20240104"
+#define SSH_VERSION_FREEBSD	"CheriBSD-20240701"
```
Comment (brooksdavis) [inline crypto/openssh/version.h:8]: I'm not sure this is the best value as it's probably the wrong one for older releng branches which will also need patching. I think there's probably no good option

Comment (kwitaszczyk) [reply crypto/openssh/version.h:8]: I wonder if this will require patching some third-party software that detects a remote OS.

**Review by kwitaszczyk — APPROVED** [review]

File: crypto/openssh/version.h (hunk 1 above)
Comment (jrtc27) [reply crypto/openssh/version.h:8]: The 14.0 patch bumps it to this version too, which seems ill-advised, but means we'd be consistent to use FreeBSD-...

Comment (jrtc27) [reply crypto/openssh/version.h:8]: And the 13.3 patch bumps from this exact version: https://security.FreeBSD.org/patches/SA-24:04/openssh-13.3.patch

Comment (brooksdavis) [reply crypto/openssh/version.h:8]: I'll switch back to FreeBSD-

**Pull request merged**